math/fibonacci_matrix_exponentiation.cpp

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file fibonacci_matrix_exponentiation.cpp
+ * @brief Computes N^th Fibonacci number given as
+ * input argument .
+ *
+ * Takes O(logn) time to compute nth Fibonacci number
+ * 
+ *
+ * \author [villayatali123](https://github.com/villayatali123)
+ * \author [unknown author]()
+ * @see fibonacci.cpp, fibonacci_fast.cpp, string_fibonacci.cpp, fibonacci_large.cpp
+ */
+
+#include<iostream>
+#include<vector>
+#include <cassert>
+
+/**
+ * This function finds nth fibonacci number.
+ * @param n nth fibonacci number
+ * @param mod  modulo number to avoid overflow
+ */
+int fibo(int n , int mod )
```

Comment:
```suggestion
uint64_t fibo(uint64_t n , uint64_t mod )
```
to handle large numbers